ndcube/ndcube.py

```diff
@@ -381,24 +391,46 @@ def axis_world_coords(self, *axes):
                 if n_dependent_axes[i] == 1:
                     # Construct pixel quantities in each dimension letting
                     # other dimensions all have 0 pixel value.
-                    quantity_list = [
-                        u.Quantity(np.zeros(cube_dimensions[dependent_axes[i]]),
-                                   unit=u.pix)] * n_dimensions
                     # Replace array in quantity list corresponding to current axis with
                     # np.arange array.
-                    quantity_list[axis] = u.Quantity(np.arange(cube_dimensions[axis]), unit=u.pix)
```

Comment:
I think this can be simplified by creating a hidden function that outputs the pixel values along an axis given the value of ```edges```:
```python
def _pixel_centers_or_edges(axis_length, edges): # edges is a bool
    if edges is False:
        axis_values = np.arange(axis_length)
    else:
        axis_values = np.arange(-0.5, axis_length+0.5)
    return axis_values
```
And then replace any instance of ```np.arange(cube_dimensions[axis])``` or ```np.arange(cube_dimensions[k])``` with ```_pixel_centers_or_edges(cube_dimensions[axis], edges)```.
This way the code diff should be minimal.

Reply:
Yeah, there was some repititive code, so should be broken down into a function.